readthedocs/search/documents.py

```diff
@@ -168,5 +168,6 @@ def get_queryset(self):
             queryset
             .internal()
             .exclude(ignore=True)
+            .select_related('version', 'project')
```

Comment:
```suggestion
            .select_related('version', 'project')
            .prefetch_related(
                Prefetch(
                    'sphinx_domains',
                    queryset=SphinxDomain.objects.exclude(
                        domain='std',
                        type__in=['doc', 'label']
                    )
                )
            )
```

We can also add prefetch_related here as well but as we are passing `queryset.iterator()` to `.update()` method of `PageDocument` `.prefetch_related()` will be ignored. this is the same for `ProjectDocument.users` as well